### DISCUSSION TITLE
Latest best ways of hosting images with Next.js / Vercel apps?

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#48287] [Help]
### Summary

I have been a web developer for quite some time, and the only image hosting source that has come up again and again, yet is still not easy or cheap to use, is AWS S3. My experience with S3 is probably setting up a few dozen random sites over the years, create a bucket, and programmatically sync photos I take or images I collect to S3 through Node.js scripts, optimizing the image and creating an md5 hash of the content or whatever.

I have looked around here and there over the years at image hosting services like upload.io and filestack.com and others I can't remember off the top of my head. But they are typically much more expensive than S3, maybe 3-10x more expensive from my memory.

Flickr allows you to host 1000 images for free, and Github allows you to host up to 1GB of static assets, which you can use through the CDN for GitHub pages. But 1GB of high quality images is only a few hundred images.

So I'm wondering "what is the Next.js way of doing image uploads and hosting these days?" I don't mean so far as creating a content _uploading_ site, but just even like a game world or book website with lots of rich content and images (tens of thousands of images and videos). It starts to add up cost-wise, so I'm wondering who has broken the mold / ice and figured out a good standard way forward with image / asset hosting in general.

If it's still S3, then how are you using it? How do you plug it into your Next.js app, and how to you monitor / manage / predict the S3 bills? I know they have the caculator, but it takes some effort and creative thinking to estimate growth and usage. If it's S3 or Google Cloud or something similar, what is the best and easiest way to start permanently sharing rich images and stuff on the web, under nice URLs? That sort of stuff.

My nice photos and drawings and everything are left buried on an external hard drive, or if I'm good then uploaded to the "cloud" (like Google Drive), but these are not "publisher's mediums", these are storage mediums. You can't for example host a website off of Google Drive, it has long latency and it's against the terms. So curious what engaging ideas are in this arena, so some websites have have more than 20-50 "static assets" for just their marketing site and a few scattered but rare admin panel assets for your web app. I would like to know something that can handle a bigger scale (but not big and constantly growing like Twitter or Facebook, where it would make sense to go lower level and optimize).

Anyways, just wanted to reach out, see what y'all have to say!

**Comment by arnasofc** [comment]
Hi ,do you have a working solution for this?

**Reply by lancejpollard** [reply]
Nope, it appears Google Cloud or S3 is still the way to go, but I could be wrong still!

**Comment by MonteLogic** [comment]
I am looking into using headless WordPress as I can keep big tech's magnifying glass off of my tech stack. Also, the docs even mention headless solutions. 
https://vercel.com/guides/how-to-upload-and-store-files-with-vercel

**Comment by Beetroot11** [comment]
👋 I've been trying to utilise a home hosted FTP to sync images in a Vercel's temp folder using basic-ftp in NextJS. Seems to work well other than having to manage the temp storage doesn't exceed the 512MB it's allowed. I'll let you know how it goes, it might be slow to be honest 😂 but at least there's no big bill at the end!

**Comment by orvvro** [comment]
Have you looked into [bunny.net](https://bunny.net/)? They seem like a great option for hosting images online. I'm looking for an option myself right now, but I'll probably start with [imgix](https://www.imgix.com/) because they're like Flickr with an optimisation API - 1000 images for free. (I'd rather optimize on the hosting platform than using Next's built-in Image API).

On a side note, why would you want your images to be bigger than 1 MB on the web? Just curious... Do you want to display 10-bit images (which jpg doesn't even support)? Or do you want your images to be at least 4k? Anyhow, I recommend converting the images to WebP for texture-rich images and to AV1 (avif) for images with less texture before uploading them to hosting options even if they have built-in image optimisation. (avif is not as widely supported as of writing this comment - see [caniuse](https://caniuse.com/avif))

**Reply by TobiramaSan** [reply]
Yes, I agree with you on this.  Bunny.net is great for CDN needs, but imgix is excellent for image optimization with a free tier for 1000 images. Keeping web images under 1 MB is ideal for speed, but larger sizes might be necessary for high-res displays. Pre-convert images to WebP or AVIF for better compression and quality. This ensures faster load times and a better user experience.

**Comment by crispopp** [comment]
To host images with Next.js/Vercel apps, consider using Capra Host's robust hosting services alongside cloud storage solutions like Amazon S3. **[Capra Host](https://caprahost.com/)**'s reliable infrastructure and fast performance ensure optimal delivery of images for your Next.js/Vercel applications. Utilizing Capra Host's hosting alongside cloud storage guarantees a seamless experience for hosting images with Next.js/Vercel apps.

**Comment by tomcurran2** [comment]
Looking to optimize image hosting for your Next.js/Vercel apps? Consider using Next.js Image Component for responsive, lazy-loaded images. Integrate a CDN like Cloudflare for faster delivery. For dynamic resizing, explore external services like Cloudinary. For detailed insights, visit **[Hosting Mastery Hub](https://hostingmasteryhub.com/)**.
